src/components/SeatMap/api.test.js

```diff
@@ -0,0 +1,226 @@
+import { JetsSeatMapApiService } from './api';
+import {
+  cabin,
+  entertainment,
+  power,
+  wifi,
+  seatDetails,
+  cabinItem,
+} from './__fixtures__/seatMapApiPostDataResponse';
+import { flightDetails } from './__fixtures__/seatMapApiGetPlaneFeatures';
+import { availabilityData } from './__fixtures__/seatMapApiAvailabilityDataResponse';
+
+jest.mock('./api', () => {
+  const module = jest.requireActual('./api');
+  return {
+    ...module,
+    JetsSeatMapApiService: class extends module.JetsSeatMapApiService {
+      postData = jest.fn();
+    },
+  };
+});
+
+const api = new JetsSeatMapApiService('test', 'test', 'test');
+
+describe('JetsSeatMapApiService', () => {
+  describe('for a specific cabin class (F, B, P or E)', () => {
+    it('should throw if no seat details were returned from the API', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      await expect(api.getPlaneFeatures(flightFixture)).rejects.toThrow('data is not found for the flight: 1111');
+    });
+
+    it('should throw if the response item has an error', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+        error: 'Something went wrong!'
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      await expect(api.getPlaneFeatures(flightFixture)).rejects.toThrow('Something went wrong!');
+    })
+
+    it('should return the correct information when all the information is present', async () => {
+      const flightFixture = flightDetails({ cabinClass: 'E' });
+      const singleCabinResponseFixture = [{
+        id: '1111',
+        cabin: cabin(),
+        entertainment: entertainment(),
+        power: power(),
+        wifi: wifi(),
+        seatDetails: seatDetails(),
+      }];
+      api.postData.mockImplementation(() => singleCabinResponseFixture);
+
+      const result = await api.getPlaneFeatures(flightFixture);
+
+      expect(result).toEqual({
```

Comment:
It would be good to verify also that api.postData was called with the expected parameters